test/sentry/telemetry/span_processor_test.exs

```diff
@@ -0,0 +1,30 @@
+defmodule Sentry.Telemetry.SpanProcessorTest do
+  use Sentry.Case, async: false
+
+  import Sentry.TestHelpers
+
+  defmodule TestEndpoint do
+    require OpenTelemetry.Tracer, as: Tracer
+
+    def instrumented_function do
+      Tracer.with_span "instrumented_function" do
+        :timer.sleep(100)
+      end
+    end
+  end
+
+  test "sends captured spans as transactions" do
+    put_test_config(environment_name: "test")
+
+    Sentry.Test.start_collecting_sentry_reports()
+
+    TestEndpoint.instrumented_function()
+    assert [%Sentry.Transaction{} = transaction] = Sentry.Test.pop_sentry_transactions()
+
+    assert length(transaction.spans) == 1
+    assert [span] = transaction.spans
+    assert span.op == "instrumented_function"
+    assert span.timestamp > span.start_timestamp
+    assert transaction.timestamp > transaction.start_timestamp
+  end
+end
```

Comment:
<img width="844" alt="Screenshot 2024-09-04 at 19 22 51" src="https://github.com/user-attachments/assets/430ed2b6-3711-4663-b8c4-abeef0dca261">

@whatyouhide do you happen to know why `start_timestamp` and `timestamp` values are negative?

Reply:
Depends on your system I think but it could be that these are in the `:native` time unit. How are you creating them? You getting them from OTel?

For example, see https://hexdocs.pm/opentelemetry_api/1.3.1/opentelemetry.html#timestamp_to_nano/1 or https://hexdocs.pm/elixir/System.html#convert_time_unit/3

Reply:
> Depends on your system I think but it could be that these are in the :native time unit. How are you creating them? You getting them from OTel?

Just via Tracer:

```elixir
  def mount(_params, _session, socket) do
    Tracer.with_span "list_posts" do
      posts = Publishing.list_posts()

      Tracer.set_attribute("posts.count", length(posts))

      {:ok, stream(socket, :posts, posts)}
    end
  end
```

Reply:
Yeah getting them from OTel then. I'd try `:opentelemetry.timestamp_to_nano` on those values and see if it gives you a reasonable nanosecond Unix timestamp 😄